src/ffi/c.rs

```diff
@@ -54,7 +56,15 @@ impl Default for StreamWrapper {
                 zalloc: Some(zalloc),
                 #[cfg(not(all(feature = "any_zlib", not(feature = "cloudflare-zlib-sys"))))]
                 zfree: Some(zfree),
-            }),
+            })),
+        }
+    }
+}
+
+impl Drop for StreamWrapper {
+    fn drop(&mut self) {
+        unsafe {
+            drop(Box::from_raw(self.inner));
```

Comment:
Let's only put into unsafe what is actually `unsafe`. Alternatively, could you explain why `drop` is part of it?